src/ImplicitMSALAuthenticationProvider.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module ImplicitMSALAuthenticationProvider
+ */
+
+import { AuthenticationParameters, AuthResponse, InteractionRequiredAuthError, UserAgentApplication } from "msal";
+
+import { AuthenticationProvider } from "./IAuthenticationProvider";
+import { AuthenticationProviderOptions } from "./IAuthenticationProviderOptions";
+import { MSALAuthenticationProviderOptions } from "./MSALAuthenticationProviderOptions";
+
+/**
+ * @class
+ * Class representing ImplicitMSALAuthenticationProvider
+ * @extends AuthenticationProvider
+ */
+export class ImplicitMSALAuthenticationProvider implements AuthenticationProvider {
+	/**
+	 * @private
+	 * A member holding an instance of MSALAuthenticationProviderOptions
+	 */
+	private options: MSALAuthenticationProviderOptions;
+
+	/**
+	 * @private
+	 * A member holding an instance of MSAL UserAgentApplication
+	 */
+	private msalInstance: UserAgentApplication;
+
+	/**
+	 * @public
+	 * @constructor
+	 * Creates an instance of ImplicitMSALAuthenticationProvider
+	 * @param {UserAgentApplication} msalInstance - An instance of MSAL UserAgentApplication
+	 * @param {MSALAuthenticationProviderOptions} options - An instance of MSALAuthenticationProviderOptions
+	 * @returns An instance of ImplicitMSALAuthenticationProvider
+	 */
+	public constructor(msalInstance: UserAgentApplication, options: MSALAuthenticationProviderOptions) {
```

Comment:
I'm not sure msalInstance is the right name for this.  I think it should be called `clientApplication` or `msalApplication`.

Reply:
fixed.